core/src/test/java/com/linecorp/armeria/client/HttpClientIntegrationTest.java

```diff
@@ -729,6 +742,47 @@ void testUpgradeRequestExecutesLogicOnlyOnce() throws Exception {
         clientFactory.close();
     }
 
+    @Nested
+    @TestInstance(Lifecycle.PER_CLASS)
+    class JettyInterop {
+
+        org.eclipse.jetty.server.Server jetty;
+
+        @BeforeAll
+        void startJetty() throws Exception {
+            jetty = new org.eclipse.jetty.server.Server(0);
+            jetty.setHandler(new AbstractHandler() {
+                @Override
+                public void handle(String target, Request baseRequest, HttpServletRequest request,
+                                   HttpServletResponse response) throws IOException, ServletException {
+                    if (Collections.list(request.getHeaders("host")).size() > 1) {
```

Comment:
nit: don't we need to check if the size of the host header is just one?
```
if (Collections.list(request.getHeaders("host")).size() == 1) {
    response.setStatus(HttpStatus.OK.code());
} else {
    response.sendError(HttpStatus.INTERNAL_SERVER_ERROR.code());
}
```

Reply:
@anuraaga ^^